sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/CredentialName.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.ContainerRegistry.Models
+{
+    /// <summary> The name of the credential. </summary>
+    public readonly partial struct CredentialName : IEquatable<CredentialName>
+    {
+        private readonly string _value;
+
+        /// <summary> Initializes a new instance of <see cref="CredentialName"/>. </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="value"/> is null. </exception>
+        public CredentialName(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string Credential1Value = "Credential1";
```

Comment:
What's this?

Reply:
This is part of our [credential set](https://learn.microsoft.com/en-us/cli/azure/acr/credential-set?view=azure-cli-latest) feature. We currently only support 1 credential in a credential set. This design will allow us to support more in the future (admin credentials, token passwords which have primary and backup). The auth credentials exists in an array with currently 1 item.

Reply:
got it, thanks